### PR TITLE
update guardrails protected namespaces list

### PR DESCRIPTION
**Pull request by yjst2012**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-8531

### What this PR does / why we need it:

removed some unnecessary ns from the protected ns list to avoid potential customer issue
### Test plan for issue:

$ ./scripts/test.sh 
expand constraints gkconstraints/aro-cluster-role-deny.yaml
expand constraints gkconstraints/aro-machine-config-deny.yaml
expand constraints gkconstraints/aro-machines-deny.yaml
expand constraints gkconstraints/aro-master-toleration-pod-deny.yaml
expand constraints gkconstraints/aro-privileged-namespace-deny.yaml
expand constraints gkconstraints/aro-pull-secret-deny.yaml
expand constraints gkconstraints/aro-service-account-deny.yaml
[opa test] -> gktemplates-src/library/common.rego gktemplates-src/aro-deny-delete-pull-secret/*.rego
PASS: 4/4
[gator verify] -> gktemplates-src/aro-deny-delete-pull-secret
=== RUN   deny-pull-secret-delete-tests
    === RUN   allow-create-pull-secret
    --- PASS: allow-create-pull-secret	(0.005s)
    === RUN   allow-update-pull-secret
    --- PASS: allow-update-pull-secret	(0.003s)
    === RUN   not-allow-delete-pull-secret
    --- PASS: not-allow-delete-pull-secret	(0.004s)
--- PASS: deny-pull-secret-delete-tests	(0.016s)
ok	gktemplates-src/aro-deny-delete-pull-secret/suite.yaml	0.016s
PASS

[opa test] -> gktemplates-src/library/common.rego gktemplates-src/aro-deny-labels/*.rego
PASS: 3/3
[gator verify] -> gktemplates-src/aro-deny-labels
=== RUN   machines-deny
    === RUN   master-machine-deny
    --- PASS: master-machine-deny	(0.003s)
    === RUN   worker-machine-allowed
    --- PASS: worker-machine-allowed	(0.002s)
    === RUN   no-label-machine-allowed
    --- PASS: no-label-machine-allowed	(0.002s)
--- PASS: machines-deny	(0.011s)
ok	gktemplates-src/aro-deny-labels/suite.yaml	0.011s
PASS

[opa test] -> gktemplates-src/library/common.rego gktemplates-src/aro-deny-machine-config/*.rego
PASS: 5/5
[gator verify] -> gktemplates-src/aro-deny-machine-config
=== RUN   deny-cluster-machineconfig-modification-tests
    === RUN   not-allow-create-cluster-machine-config
    --- PASS: not-allow-create-cluster-machine-config	(0.005s)
    === RUN   not-allow-delete-cluster-machine-config
    --- PASS: not-allow-delete-cluster-machine-config	(0.003s)
    === RUN   not-allow-update-cluster-machine-config
    --- PASS: not-allow-update-cluster-machine-config	(0.004s)
    === RUN   allow-create-custom-machine-config
    --- PASS: allow-create-custom-machine-config	(0.005s)
    === RUN   allow-delete-custom-machine-config
    --- PASS: allow-delete-custom-machine-config	(0.003s)
--- PASS: deny-cluster-machineconfig-modification-tests	(0.023s)
ok	gktemplates-src/aro-deny-machine-config/suite.yaml	0.023s
PASS

[opa test] -> gktemplates-src/library/common.rego gktemplates-src/aro-deny-master-toleration-taints/*.rego
PASS: 5/5
[gator verify] -> gktemplates-src/aro-deny-master-toleration-taints
=== RUN   deny-master-toleration-taint-pods-in-nonprivileged-namespaces
    === RUN   create-not-allowed-in-nonprivileged-namespaces
    --- PASS: create-not-allowed-in-nonprivileged-namespaces	(0.004s)
    === RUN   create-allowed-in-privileged-namespaces
    --- PASS: create-allowed-in-privileged-namespaces	(0.004s)
    === RUN   update-not-allowed-in-nonprivileged-namespaces
    --- PASS: update-not-allowed-in-nonprivileged-namespaces	(0.005s)
    === RUN   deletion-allowed-in-nonprivileged-namespaces
    --- PASS: deletion-allowed-in-nonprivileged-namespaces	(0.004s)
--- PASS: deny-master-toleration-taint-pods-in-nonprivileged-namespaces	(0.021s)
ok	gktemplates-src/aro-deny-master-toleration-taints/suite.yaml	0.021s
PASS

[opa test] -> gktemplates-src/library/common.rego gktemplates-src/aro-deny-privileged-namespace/*.rego
PASS: 16/16
[gator verify] -> gktemplates-src/aro-deny-privileged-namespace
=== RUN   privileged-namespace
    === RUN   ns-allowed-pod
    --- PASS: ns-allowed-pod	(0.004s)
    === RUN   ns-disallowed-pod
    --- PASS: ns-disallowed-pod	(0.005s)
    === RUN   ns-disallowed-deploy
    --- PASS: ns-disallowed-deploy	(0.004s)
    === RUN   ns-allowed-deploy
    --- PASS: ns-allowed-deploy	(0.004s)
--- PASS: privileged-namespace	(0.023s)
ok	gktemplates-src/aro-deny-privileged-namespace/suite.yaml	0.023s
PASS

[opa test] -> gktemplates-src/library/common.rego gktemplates-src/library/*.rego
[gator verify] -> gktemplates-src/library
PASS

go test passed

### Is there any documentation that needs to be updated for this PR?

[the guardrails PND doc](https://docs.google.com/document/d/19cKNlAGRte1bUBgAgFDEe6l7MiULgRqmOu-N12jQW04/edit) updated

### How do you know this will function as expected in production? 

the logic is quite simple and straightforward

### CODE REVIEW

**Review by hlipsig — APPROVED** [review]

File: pkg/operator/controllers/guardrails/policies/gktemplates-src/library/common.rego [hunk 1]
```diff
@@ -113,31 +108,35 @@ privileged_ns = {
...
+  # "openshift-marketplace",
+  # "openshift-ingress-canary",
+  # "openshift-insights",
+  # "openshift-kni-infra",
+  # "openshift-cluster-csi-drivers",
+  # "openshift-cluster-node-tuning-operator",
+  # "openshift-cluster-samples-operator",
+  # "openshift-config",
+  # "openshift-ovirt-infra",
+  "openshift-sdn"
```
Comment (bitoku) [inline pkg/operator/controllers/guardrails/policies/gktemplates-src/library/common.rego:141]: It would be more readable to make it in alphabetical order?

**Review by bitoku — APPROVED** [review]

File: pkg/operator/controllers/guardrails/policies/gktemplates-src/library/common.rego (hunk 1 above)
Comment (yjst2012) [reply pkg/operator/controllers/guardrails/policies/gktemplates-src/library/common.rego:141]: these removed items were temporarily put here, once confirmed they will be removed from code base as this file will be included in many other policies.

**Review by jaitaiwan — APPROVED** [review]

**Pull request merged**